### ISSUE TITLE
vscode incorrectly report empty 'files' list on tsconfig with 'references' field

### ISSUE THREAD

**Issue by jjyyxx** [Microsoft/vscode#58697]
- VSCode Version: 1.27.2
- OS Version: Windows 10 1803
- Typescript Version: 3.0.3 (both the vsc bundled ts and the local ts)

## Description

The `tsconfig.json` with content
```json
{
  "references": [
    { "path": "./packages/foo" },
    { "path": "./packages/bar" }
  ],
  "files": []
}
```
reports the problem `[ts] The 'files' list in config file '/path/to/tsconfig.json' is empty.` which is incorrect.
![image](https://user-images.githubusercontent.com/13808089/45556145-25355b80-b86d-11e8-8b03-32aa65a58385.png)

## Steps to Reproduce

*The problem is rather strange and not so easy to reproduce. I am even not so sure that it's a bug with vscode or one with typescript.*
**But I will try to detailly explain the current behavior. I would like to provide more infomation if given some help or hint or some debugging techniques to help me dig further.**

- The problem report does not show up on starting up or simply opening the `tsconfig.json`, even after waiting for 30 minutes. All other tsconfig checks doesn't perform too (e.g. set `files` field to "" does not raise problem at that time)
- But if I made some arbitrary edits in some `.ts` files, save it *(not so confident about this step as sometimes it shows up quickly and sometimes takes a long time)* and wait for some time (around 10s), such problem report will appear and all other checks begin to function normally.
- If I add some entries in the `files` field, such problem report will disappear in around 10s. 
- If I remove those entries, such problem will reappear in around 10s. 

So the overall behavior reveals **extreme latency**.

Does this issue occur when all extensions are disabled?: Yes

**Comment by jjyyxx** [comment]
I found that the problem raises too if I run the tsc:build task.
![image](https://user-images.githubusercontent.com/13808089/45556673-6843fe80-b86e-11e8-94e4-6d5150a58175.png)

As currently vscode's `typescript-language-features` extension does not take references into consideration (I made a pr to fix that) and always use `-p` flag, it justifies such problem report. So maybe some check services performed on `tsconfig.json` always use `-p` flag too.
